php-sync/sfs-sync.php

```diff
@@ -305,40 +306,32 @@ public function scheduleBatches ($mode, $toSchedule) {
 
 			$dir = $this->config["BATCHDIR"]."/$mode/$node";
 			if (!is_dir ($dir)) {
-				continue;
+				mkdir ($dir, 0777, TRUE);
 			}
-			
-			$hBatchDir = opendir($dir);
-			if ($hBatchDir === FALSE) {
+			if(!is_dir($dir)){
 				syslog(LOG_CRIT, "Cannot opendir $dir, will retry in ".$this->config["FAILTIME"]." seconds");
 				$this->setFailing ($node);
-				$tasks[] = array ();
+				continue;
+			}
+
+			$allBatches = glob($dir . '/*_*.batch');
```

Comment:
At this point we schedule the transfer of data. If we don't want inconsistency, we have to ensure a global sorting of the batches across all nodes. Or am I wrong?

But in any case we get inconsistent if file A is changed in this order
node2/A
node3/A
node2/A

in one transition. If we want to have the correct result, the files need to be read of all directories and sorted in reverse order, so the newest file on all nodes succeeds.